vmhost/contexts/async.go

```diff
@@ -150,14 +154,15 @@ func (context *asyncContext) InitStateFromInput(input *vmcommon.ContractCallInpu
 // internal state stack.
 func (context *asyncContext) PushState() {
 	newState := &asyncContext{
-		callID:          context.callID,
-		callerCallID:    context.callerCallID,
-		callerAddr:      context.callerAddr,
-		callback:        context.callback,
-		callbackData:    context.callbackData,
-		gasAccumulated:  context.gasAccumulated,
-		returnData:      context.returnData,
-		asyncCallGroups: context.asyncCallGroups, // TODO matei-p use cloneCallGroups()?
+		originalCallerAddr: context.originalCallerAddr,
```

Comment:
This needs to be added to `PopSetActiveState()` as well.

Reply:
And to `Clone()`.

Reply:
fixed